DAL/Repositories/Interfaces/IOrderRepository.cs

```diff
@@ -0,0 +1,10 @@
+﻿using DAL.Entities;
+
+namespace DAL.Repositories.Interfaces;
+
+public interface IOrderRepository
+{
+	Task<IEnumerable<Order>> GetOrdersByUserAsync(User user);
+	Task<IEnumerable<Product>> GetProductsByOrderIdAsync(int orderId);
```

Comment:
it is better to put the method in the base repository like GetBySmth and use predicate